modules/claims_api/app/controllers/claims_api/v1/forms/disability_compensation_controller.rb

```diff
@@ -190,6 +191,44 @@ def validate_form_526_location_codes!
           end
         end
 
+        def validate_form_526_service_information_confinements!
+          confinements = form_attributes['serviceInformation']['confinements']
+          service_periods = form_attributes['serviceInformation']['servicePeriods']
+
+          return if confinements.nil?
+
+          return if confinements_within_service_periods?(confinements,
+                                                         service_periods) && confinements_dont_overlap?(confinements)
+
+          raise ::Common::Exceptions::InvalidFieldValue.new('confinements', confinements)
+        end
+
+        def confinements_within_service_periods?(confinements, service_periods)
+          confinements.each do |confinement|
+            next if service_periods.any? do |service_period|
+              time_range = service_period['activeDutyBeginDate']..service_period['activeDutyEndDate']
+              time_range.cover?(confinement['confinementBeginDate']) &&
+              time_range.cover?(confinement['confinementEndDate'])
```

Comment:
Should we maybe convert these `xBeginDate` & `xEndDate` values into actual Date objects and then build our ranges?
Right now, we'll get a range of `String` values and it will include values like `1984-00-00` and `1984-99-99`.

So something like this would be possible.
```
range1 = '1980-03-05'..'1985-01-07'
range1.include?('1984-66-99')  #will be true
```

Just wondering if we need to protect against that by making our ranges be date based.

Reply:
good catch. I was doing that conversion in the request spec, but I should be doing that here, instead

Reply:
Addressed [here](https://github.com/department-of-veterans-affairs/vets-api/pull/7661/commits/c8daeded90a3e781f7a6022b7333f7957b11a52b